blog/2024-11-27-flet-v-0-25-release-announcement.md

```diff
@@ -0,0 +1,294 @@
+
+* `FLET_APP_DATA` - directory for storing application data that is preserved between app updates. That directory is already pre-created and its location depends on the platform the app is running on.
+* `FLET_APP_TEMP` - directory for temporary application files, i.e. cache. That directory is already pre-created and its location depends on the platform the app is running on.
+
+For example, data folder path can be read in your app as:
+
+```
+import os
+
+# it's `None` when running the app in web mode
+os.getenv("FLET_APP_DATA")
+```
+
+## `pyproject.toml` support
+
+It's inconvenient and bulky to carry all `flet build` settings as command line options.
+
+In Flet 0.25.0 release is now possible to configure `flet build`, `flet run`, and `flet publish` settings in `pyproject.toml`!
+
+Flet adds `tool.flet` section and sub-sections to `pyproject.toml`. Here's the minimal version of `pyproject.toml` you can put into root folder of your app:
+
+```toml
+[project]
+name = "my_app"
+version = "1.0.0"
+description = "My first Flet project"
+authors = [
+  {name = "John Smith", email = "john@email.com"}
+]
+dependencies = ["flet"]
+```
+
+You can also generate a starting `pyproject.toml` (and other files) for your project with `flet create` command.
+
+
+## New Python packages structure
+
+We re-factored Flet Python packages and removed `flet-core`, `flet-runtime`, `flet-embed` or`flet-pyodide` packages.
+
+The new structure avoids rewriting pip dependencies while installing `flet` package on various platforms. There was a problem of detecting the correct `flet` package to install (`flet-runtime`, `flet-embed` or`flet-pyodide`?) if `flet` was not a direct dependency in user's app.
+
+New Flet packages:
+
+* `flet` - required for minimal Flet setup, app entry point for various platforms with core logic and controls. Installed on all platforms.
+* `flet-cli` - contains Flet CLI commands. Installed on desktop only.
+* `flet-desktop` - contains pre-built Flet "client" app binary for macOS, Windows and Linux. By default installed on macOS and Windows desktops only. 
+* `flet-desktop-light` - contains a light-weight version (without Audio and Video controls) of Flet "client" for Linux. By default installed on Linux desktops only. 
+* `flet-web` - contains Flet web "client" and FastAPI integration. Installed on desktop only.
+
+Packaged Flet app contains only `flet` package now.
+
+### `flet` package extras
+
+`flet` package defines the following extras that can be specified when installing Flet with pip, uv, poetry and other package manager:
+
+* `flet[all]` - installs `flet`, `flet-cli`, `flet-desktop` and `flet-web`. Recommended for development.
+* `flet[cli]` - installs `flet` and `flet-cli`. Can be used in CI environment for packaging only.
+* `flet[web]` - installs `flet` and `flet-web`. Used for deploying Flet web apps.
+* `flet[desktop]` - installs `flet` and `flet-desktop`. Used for desktop-only development.
```

Comment:
what about flet-desktop-light? will be nice to mention when it is installed and how to switch to flet-desktop (with audio and video), for those with no/resolved linux issues.